### PR TITLE
test: update confusing e2e democracy setup

### PR DESCRIPTION
**Pull request by MSalopek**
## Description

Democracy and democracy reward e2e tests were wired in a confusing way. This PR removes the confusion.

Before the PR, `democracy` tests actually ran `democracy-reward` tests and vice versa. It was difficult to work with the tests since there are configuration nuances that are easy to overlook with regards to reward distribution tests.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Comment by p-offtermatt** [comment]
Suggestion: Since the two steps steps_democracy and steps_reward_denom only differ in exactly the true/false statements, could we make these steps a single function that just takes a boolean as an input to reduce code duplication?
Also, could we rename them to more evocative names? The only difference seems to be whether the denom is enabled or not. Could we call one democracy_missing_denom and the other democracy_ok_denom?

**Comment by MSalopek** [comment]
> Suggestion: Since the two steps steps_democracy and steps_reward_denom only differ in exactly the true/false statements, could we make these steps a single function that just takes a boolean as an input to reduce code duplication? Also, could we rename them to more evocative names? The only difference seems to be whether the denom is enabled or not. Could we call one democracy_missing_denom and the other democracy_ok_denom?

We can, I'm doing this to unblock some of the v50 work

**Review by p-offtermatt — APPROVED** [review]
LGTM!

File: tests/e2e/steps_democracy.go [hunk 1]
```diff
@@ -181,12 +181,13 @@ func stepsDemocracy(consumerName string) []Step {
 			},
 			State: State{
 				ChainID("provi"): ChainState{
-					// Check that tokens are minted and sent to provider chain and distributed to validators and their delegators on provider chain
+					// Check that ARE NOT minted and sent to provider chain and distributed to validators and their delegators on provider chain
```
Comment (bermuell) [inline tests/e2e/steps_democracy.go:184]:
```suggestion
					// Check that tokens ARE NOT minted and sent to provider chain and distributed to validators and their delegators on provider chain
```

**Pull request merged**